cfme/utils/appliance/__init__.py

```diff
@@ -1726,13 +1726,18 @@ def server_roles(self, roles):
         if self.server_roles == roles:
             self.log.debug(' Roles already match, returning...')
             return
-        ansible = roles.get('embedded_ansible', False)
+        ansible_old = self.server_roles('embedded_ansible', False)
+        ansible_new = roles.get('embedded_ansible', False)
+        enabling_ansible = False
+        if ansible_old is False and ansible_new is True:
+            enabling_ansible = True
+
         yaml = self.get_yaml_config()
         yaml['server']['role'] = ','.join([role for role, boolean in roles.iteritems() if boolean])
         self.set_yaml_config(yaml)
-        timeout = 600 if ansible else 300
+        timeout = 900 if enabling_ansible else 300
```

Comment:
So if I understand correctly, the UI role is enabled almost immediately, but the workers for the role are what take several minutes to start up.

If my understanding is correct, can't we leave this timeout to be 300 for the UI update, and apply the longer timeout to `self.wait_for_embedded_ansible` call on L1741

Reply:
Good catch! Yes, I agree with this, but Pete previously increased this to 600 and after talking to @quarckster I thought even 900 might be needed. But your are correct that 900 is really needed for `wait_for_embedded_ansible`, not for this. And `wait_for_embedded_ansible` is set to 900 by default so... I'll change this one back to 600, assuming Pete had a better reason to increase it to 600 than I have for the 900.

Reply:
@psav can you confirm behavior here, UI doesn't actually need 600s to update roles, just the workers?  Ultimately if that's the case the timeout will never actually come into play anyway :)